lib/components/data/tables/CWMSTable.jsx

```diff
@@ -0,0 +1,180 @@
+            throw Error("You must specify one or more Timeseries IDs to table.");
+
+        if (!office) throw Error("You must specify a 3 letter ID for the office");
+
+        const fetchData = async () => {
+            let promises = tsids.map(async (name) => {
+                try {
+                    return await ts_api.getCwmsDataTimeseries({
+                        name,
+                        begin,
+                        end,
+                        pageSize,
+                        unit,
+                        office,
+                    });
+                } catch (error) {
+                    if (error.response?.status === 404) {
+                        console.warn(`Data for ${name} not found: 404`);
+                        return null;
+                    } else {
+                        throw error;
+                    }
+                }
+            });
+
+            let values = await Promise.all(promises);
+            let _data = { ts: {}, dates: [] };
+            let dates = []
+            if (!interval) { interval = 1 }
+
+            values.forEach((result) => {
+                if (result && result.values) {
+                    if (!_data.ts[result.values]) {
+                        _data.ts[result.name] = [];
+                    }
+                    _data.ts[result.name].push(result);
+                    result.values.forEach((item) => {
+                        const dt = item[0]
+                        const val = item[1]
+                        if (!dates.includes(dt) && val != null) {
+                            if (dt % (interval * 1000 * 60) == 0) {
+                                dates.push(dt)
+                            }
+                        }
+                    })
+                } else if (result === null) {
+                    console.warn(`Skipping as no data was found.`);
+                } else {
+                    console.warn(`No unit found for ${result?.name}`);
+                }
+            });
+            dates.sort()
+
+            if (order) {
+                if (order == "desc") { dates.reverse() }
+            }
+
+            _data.dates = dates
+
+            setTsData(_data)
+
+        };
+
+        fetchData();
+
+    }, [])
+
+
+    useEffect(() => {
+        if (!tableElement.current || !tsData) {
+            return;
+        }
+        tsData.dates.forEach((dt) => {
+            const row = []
+            Object.keys(tsData.ts).forEach((item) => {
+                let rounding = 2
+                Object.entries(timeseriesParams).map(entry => {
+                    if (entry[1].tsid == item && entry[1].precision != null) {
+                        rounding = entry[1].precision
+                    }
+                })
+                tsData.ts[item][0].values.forEach((val) => {
+                    (dt == val[0]) && row.push(
+                        val[1] ? val[1].toFixed(rounding) : null
+                    )
```

Comment:
The way the time series data is being inserted leads to an erroneous table if a time series has a shorter interval than a preceding time series:
![image](https://github.com/user-attachments/assets/18aa156a-da14-4c4e-854e-9030fdcb3a5a)
Could probably push an empty value when the date doesn't exist for a particular time series (probably not the missingString since the data isn't technically "missing").  Maybe darken the cell as well as an indicator?  We can figure out the UI specifics later as long as we get the values fixed now.